src/app/beer_garden/scheduler.py

```diff
@@ -358,6 +358,28 @@ def remove_job(self, job_id, **kwargs):
         else:
             self._sync_scheduler.remove_job(job_id, **kwargs)
 
+    def execute_job(self, job_id, reset_interval=False, **kwargs):
+        """Executes the job ad-hoc
+
+        Args:
+            job_id: The job id
+            reset_interval: Whether to set the job's interval begin time to now
+        """
+        job = db.query_unique(Job, id=job_id)
+        self.add_job(
```

Comment:
2 thoughts here:

* I wonder / worry about any side effects associated with adding a job to the scheduler that isn't persisted in the jobstore. We wouldn't want this persisted in the jobstore, but I'm curious if the schedulers internal state being out of sync with the database record of things is going to cause us headaches at some point.

* Did you explore if it's viable to just call `run_job` instead?  The docstring suggests that `run_job` is intended to be run in a separate process, but it might be worth understanding why that is (I'm assuming some async / blocking related concerns) and see if we can work around those concerns.

Reply:
To your other point, since the ad-hoc jobs have a one-time `DateTime` interval, they shouldn't be able to run afterwards. However, there is no debug message (form APscheduler, not BG) saying the job is *removed* like when deleting a normal job through the UI. Since the ad-hoc jobs would all have the same id of `None`, I believe they are being overwritten as more jobs are executed ad-hoc, meaning not much clutter. There may be a way to delete jobs after they're run via APScheduler that I'm not aware of.